helm/resource_release.go

```diff
@@ -400,7 +402,23 @@ func resourceRelease() *schema.Resource {
 	}
 }
 
-func resourceReleaseRead(d *schema.ResourceData, meta interface{}) error {
+func resourceReleaseRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	exists, err := resourceReleaseExists(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if !exists {
+		return diag.Diagnostics{
+			{
+				Severity: diag.Warning,
+				Summary:  "Release does not exist",
+				Detail:   "resourceReleaseExists() returned 'false' for resource ID: " + d.Id(),
```

Comment:
I don't know if codifying internal function names in diagnostic details is a good idea or if that's even helpful to the end user. I think diagnostics are for pointing the user to problems with their configuration, not problems inside the provider code. e.g. Highlighting that a particular attribute is misconfigured in some way.

After the call above we spit out the error if the function returns error, so here we know that in this case the release simply does not exist. Perhaps we could tell them why they might encounter this situation, i.e that it was deleted manually.